eng/pipelines/templates/steps/verify-autorest.yml

```diff
@@ -0,0 +1,66 @@
+parameters:
+  - name: ServiceToTest
+    type: string
+    default: ''
+  - name: SDKType
+    type: string
+    default: 'client'
+  - name: ServiceDirectory
+    type: string
+  - name: NodeVersion
+    type: string
+    default: '12.x'
+  - name: PythonVersion
+    type: string
+    default: '3.9'
+  - name: auto_rest_clone_url
+    type: string
+    default: 'https://github.com/Azure/autorest.python.git'
+
+steps:
+  - task: UsePythonVersion@0
+    displayName: Use
+    inputs:
+      versionSpec: ${{ parameters.PythonVersion }}
+
+  - task: NodeTool@0
+    inputs:
+      versionSpec: ${{ parameters.NodeVersion }}
+
+  - script: |
+      npm install -g autorest
+      autorest --help
+    displayName: "Install autorest"
+
+  - script: |
+      python --version
+      python -m pip install pip==20.1
+      pip install -r eng/autorest_req.txt
+      git clone ${{ parameters.auto_rest_clone_url }}
+      cd autorest.python
+      npm install
+      cd ..
+    displayName: 'Prepare Environment'
+
+  - pwsh: |
+      Get-ChildItem ./
+    displayName: 'List contents'
+    condition: always()
+
+  - task: PythonScript@0
+    displayName: 'Run autorest Python'
+    inputs:
+      scriptPath: 'scripts/devops_tasks/verify_autorest.py'
+      arguments: >-
+        --service_directory="${{ parameters.ServiceDirectory }}"
+
+  - pwsh: |
```

Comment:
Feel free to rip this entire step out. We get a flattened agent with every new run, so you don't need to spend time cleaning up.